### PR TITLE
getkeypool: --all option for descriptor wallets

### PR DESCRIPTION
**Pull request by instagibbs**
needs tests, putting this up for concept ACKs first. Seems to work(TM). 

resolves https://github.com/bitcoin-core/HWI/issues/315

### CODE REVIEW

**Comment by achow101** [comment]
Concept ACK

File: hwilib/commands.py [hunk 1]
```diff
@@ -108,7 +106,7 @@ def getkeypool_inner(client, path, start, end, internal=False, keypool=True, acc
     this_import['watchonly'] = True
     return [this_import]
 
-def getdescriptor(client, master_xpub, testnet=False, path=None, internal=False, sh_wpkh=False, wpkh=True, account=0, start=None, end=None):
+def getdescriptor(client, master_xpub, testnet=False, path=None, internal=False, addr_type="wpkh", account=0, start=None, end=None):
```
Comment (achow101) [inline hwilib/commands.py:109]: Maybe use an enum instead of strings?

Comment (instagibbs) [reply hwilib/commands.py:109]: yep

**Comment by instagibbs** [comment]
Pushed a basic test, no QT support currently because I can't be bothered.

File: hwilib/commands.py [hunk 2]
```diff
@@ -157,21 +165,40 @@ def getdescriptor(client, master_xpub, testnet=False, path=None, internal=False,
...
+    return Descriptor(master_fpr, path_base.replace('m', ''), client.xpub_cache.get(path_base), path_suffix, client.is_testnet, is_sh_wpkh, is_wpkh)
+
+def getkeypool(client, path, start, end, internal=False, keypool=True, account=0, sh_wpkh=False, wpkh=True, addr_all=False):
 
-# wrapper to allow both internal and external entries when path not given
-def getkeypool(client, path, start, end, internal=False, keypool=True, account=0, sh_wpkh=False, wpkh=True):
+    if sh_wpkh and wpkh:
+        return {'error': 'Both `--wpkh` and `--sh_wpkh` can not be selected at the same time.', 'code': BAD_ARGUMENT}
+    if addr_all and (sh_wpkh or wpkh or (path is not None) or internal):
+        return {'error': 'Both `--all` and specific address/change types can not be selected at the same time.', 'code': BAD_ARGUMENT}
```
Comment (achow101) [inline hwilib/commands.py:175]: Instead of having the checks like this and requiring multiple bool args, we should just add a mutually exclusive group to the argument parser itself. That should do what we want and should be able to get it to give us the correct enum too.

Comment (instagibbs) [reply hwilib/commands.py]: Still punting enum list construction until `getkeypool` internally, but otherwise done(I think?)

File: hwilib/commands.py [hunk 3]
```diff
@@ -180,9 +202,9 @@ def getdescriptors(client, account=0):
 
     for internal in [False, True]:
         descriptors = []
-        desc1 = getdescriptor(client, master_fpr=master_fpr, testnet=client.is_testnet, internal=internal, sh_wpkh=False, wpkh=False, account=account)
```
Comment (achow101) [inline hwilib/commands.py:183]: `master_xpub` is an unrelated change. You may need to rebase.

Comment (instagibbs) [reply hwilib/commands.py:183]: should be fixed, was an artifact of rebasing incorrectly onto master

**Comment by achow101** [comment]
ACK 15b213a6dc0e9193fc624973b22a411723cf1974
